### PR TITLE
🎉 Compare versions tool

### PR DESCRIPTION
**Pull request by lazebnyi**
## What
To compare in memory records outputs need some tool

## How
Created a bash script that pull the images for the two docker versions and calculated the difference of the record output

### CODE REVIEW

**Review by evantahler — COMMENTED** [review]
Nice!  

✔️ This keeps records in memory
✔️ It compares the output

Some docs might be nice, e.g. a readme or a comment at the top
* Looks like this needs to be run from the root dir of the connector (where /secrets is relatively nearby)
* You probably don't want to run a whole sync, so some thought should be used when choosing a starting state?  Or, maybe you do want to run a "full" sync

It also might be helpful to use a visual diff tool.  [Here are some](https://stackoverflow.com/questions/1991248/any-visual-diff-in-linux-console).

File: tools/internal/compare_versions.sh [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+else
+    state_status="without"
+    state='" "'
+fi
+
+
+printf "\nPulling connectors images:\n"
+
+docker pull airbyte/"$connector":"$version_1"
+docker pull airbyte/"$connector":"$version_2"
```
Comment (evantahler) [inline tools/internal/compare_versions.sh:39]: Is there ever a scenario where you would be testing this on a pre-release (#dev) version?  If so, docker-pull won't work.  I think `docker run` will pull the image if needed, so you might be able to skip these

File: tools/internal/compare_versions.sh [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+
+printf "\nStarting comparing connector %s version %s with version %s %s state:\n" "$connector" "$version_1" "$version_2" "$state_status"
+
+run_docker_image()
+{
+  docker run --rm -v $(pwd)/config_files:/config_files airbyte/"$connector":"$1" read --config /config_files/secrets/config.json --catalog /config_files/configured_catalog.json --state "$state"
+}
+
+result=$( diff <(run_docker_image "$version_1") <(run_docker_image "$version_2") | grep 'RECORD' | sed '/\"emitted_at\"/,/}/ d')
```
Comment (evantahler) [inline tools/internal/compare_versions.sh:49]: `diff` - neat!

**Review by evantahler — APPROVED** [review]

File: tools/internal/compare_versions.sh (hunk 2 above)
Comment (sherifnada) [reply tools/internal/compare_versions.sh:49]: should we also compare STATE messages?

**Review by sherifnada — CHANGES_REQUESTED** [review]
should we also be comparing STATE messages? it's an important part of the behavior

File: tools/internal/compare_versions.sh (hunk 1 above)
Comment (lazebnyi) [reply tools/internal/compare_versions.sh]: shifted pull to run stage

File: tools/internal/compare_versions.sh (hunk 2 above)
Comment (lazebnyi) [reply tools/internal/compare_versions.sh]: Updated

**Comment by YowanR** [comment]
@lazebnyi Please take a look at the latest comments and let's aim to wrap this up on Monday, please 🙏

**Comment by lazebnyi** [comment]
@YowanR thanks. It look like I already fixed few day ago) - https://github.com/airbytehq/airbyte/pull/24421#discussion_r1147578631

**Review by sherifnada — APPROVED** [review]

**Pull request merged**